src/checks/spatial/buddy_check.rs

```diff
@@ -1,4 +1,29 @@
-use crate::{util, Error, Flag, SpatialCache};
+use crate::{
+    util::{self, spatial_tree::SpatialTree, SingleOrVec},
+    DataCache, Error, Flag,
+};
+
+/// Specific arguments to buddy_check, broken into a struct to make the function
+/// signature more readable.
+#[derive(Debug, Clone)]
+pub struct BuddyCheckArgs {
+    /// Search radius in which buddies of an observation will be found. Unit: m
+    pub radii: SingleOrVec<f32>,
+    /// The minimum buddies an observation can have (lest it be flagged [`Flag::Isolated`])
+    pub nums_min: SingleOrVec<u32>,
```

Comment:
```suggestion
    /// Search radius in which buddies of an observation will be found. Unit: m
    pub radii: SingleOrVec<f32>,
    /// The minimum buddies an observation can have (lest it be flagged [`Flag::Isolated`])
    pub min_buddies: SingleOrVec<u32>,
```
Buddies sounds cool :smile: But why would these be `SingleOrVec` and not just the inner type (assuming we are QC'ing obses from the same timeseries)? Or are we also QC'ing the backing data?

Reply:
Clarified in conversation